server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/objectstore/BlobStoreChoosingModuleTest.java

```diff
@@ -0,0 +1,75 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.james.modules.objectstorage.FakePropertiesProvider;
+import org.apache.james.modules.objectstore.BlobStoreChoosingConfiguration.BlobStoreImplName;
+import org.apache.james.modules.objectstore.BlobStoreChoosingModule.CassandraBlobStoreFactory;
+import org.apache.james.modules.objectstore.BlobStoreChoosingModule.SwiftBlobStoreFactory;
+import org.junit.jupiter.api.Test;
+
+class BlobStoreChoosingModuleTest {
+
+    private static CassandraBlobStoreFactory CASSANDRA_BLOBSTORE_FACTORY = mock(CassandraBlobStoreFactory.class);
+    private static SwiftBlobStoreFactory SWIFT_BLOBSTORE_FACTORY = mock(SwiftBlobStoreFactory.class);
+
+    @Test
+    void provideBlobStoreFactoryShouldThrowWhenNoFile() {
+        BlobStoreChoosingModule module = new BlobStoreChoosingModule();
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register("other_configuration_file", new PropertiesConfiguration())
+            .build();
+
+        assertThatThrownBy(() -> module.provideBlobStoreFactory(propertyProvider, CASSANDRA_BLOBSTORE_FACTORY, SWIFT_BLOBSTORE_FACTORY))
+            .isInstanceOf(RuntimeException.class);
+    }
+
+    @Test
+    void provideBlobStoreFactoryShouldReturnSwiftFactoryWhenConfigurationImplIsSwift() throws Exception {
+        BlobStoreChoosingModule module = new BlobStoreChoosingModule();
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        configuration.addProperty("objectstore.implementation", BlobStoreImplName.SWIFT.getName());
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register(BLOBSTORE_CONFIGURATION_NAME, configuration)
+            .build();
+
+        assertThat(module.provideBlobStoreFactory(propertyProvider, CASSANDRA_BLOBSTORE_FACTORY, SWIFT_BLOBSTORE_FACTORY))
+            .isEqualTo(SWIFT_BLOBSTORE_FACTORY);
+    }
+
+    @Test
+    void provideBlobStoreFactoryShouldReturnCassandraFactoryWhenConfigurationImplIsCassandra() throws Exception {
+        BlobStoreChoosingModule module = new BlobStoreChoosingModule();
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        configuration.addProperty("objectstore.implementation", BlobStoreImplName.CASSANDRA.getName());
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register(BLOBSTORE_CONFIGURATION_NAME, configuration)
+            .build();
+
+        assertThat(module.provideBlobStoreFactory(propertyProvider, CASSANDRA_BLOBSTORE_FACTORY, SWIFT_BLOBSTORE_FACTORY))
+            .isEqualTo(CASSANDRA_BLOBSTORE_FACTORY);
+    }
+}
```

Comment:
You are missing some test cases.

What happens when...

 - `objectstore.implementation` is missing ?
 - `objectstore.implementation` is bad, ie value is `gabouzomeuh` ?